magma/circuit.py

```diff
@@ -758,6 +763,15 @@ def is_definition(self):
     def instances(self):
         return self._context_.placer.instances()
 
+    def get_instance(self, name: str) -> 'AnonymousCircuitType':
+        instances = self.instances
+        try:
+            return only(filter(lambda i: i.name == name, instances))
+        except EmptyIterableException:
+            raise KeyError(name) from None
+        except NonSingletonIterableException:
+            raise KeyError(f"Found multiple instances with name '{name}'")
```

Comment:
Would these get "uniquified" in the downstream verilog? Otherwise should we make this illegal in magma? Then we could consider changing the instances container to be a map.  Alternatively, we could uniquify the names at the magma level to avoid this issue?

Reply:
As of now, nothing in the magma representation requires the constraint that instances have unique names. In fact, it is even valid MLIR to have duplicate instance names, e.g.

```mlir
    hw.module @Bar(%I: i1, %CLK: i1) -> (O: i1) {
        %0 = hw.instance "foo" @Foo(I: %I: i1) -> (O: i1)
        %1 = hw.instance "foo" @Foo(I: %0: i1) -> (O: i1)
        hw.output %1 : i1
    }
```